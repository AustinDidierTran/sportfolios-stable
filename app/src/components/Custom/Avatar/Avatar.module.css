.avatar {
  font-size: 24px;
  width: 48px;
  height: 48px;
  border: 4px solid #f5f5f5;
  background-color: #18b393 !important;
  color: #fff !important;
  text-align: center;
}

.sm {
  font-size: 16px;
  width: 36px !important;
  height: 36px !important;
  margin-right: auto;
  margin-left: auto;
}

.md {
  font-size: 24px;
  width: 48px !important;
  height: 48px !important;
  margin-right: auto;
  margin-left: auto;
}

.lg {
  font-size: 24px;
  width: 72px !important;
  height: 72px !important;
  margin-right: auto;
  margin-left: auto;
}

@media (min-width: 600px) {
  .md {
    font-size: 24px !important;
    width: 64px !important;
    height: 64px !important;
  }
  .lg {
    font-size: 36px !important;
    width: 96px !important;
    height: 96px !important;
  }
}

@media (min-width: 992px) {
  .lg {
    font-size: 48px !important;
    width: 128px !important;
    height: 128px !important;
  }
}
