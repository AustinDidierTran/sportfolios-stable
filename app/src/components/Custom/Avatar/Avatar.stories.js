// YourComponent.stories.js

import React from 'react';
import Avatar from './index';

// This default export determines where you story goes in the story list
export default {
  title: 'Avatar',
  component: Avatar,
  argTypes: {
    size: {
      control: {
        type: 'inline-radio',
        options: ['sm', 'md', 'lg'],
      },
    },
    variant: {
      control: {
        type: 'inline-radio',
        options: ['circle', 'square'],
      },
    },
    initials: {
      control: {
        type: 'inline-radio',
        options: [null, 'SF'],
      },
    },
  },
};

const Template = args => <Avatar {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  size: 'md',
  variant: 'circle',
  initials: null,
};
