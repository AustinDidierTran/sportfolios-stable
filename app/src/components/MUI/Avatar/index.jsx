import React from 'react';
import Avatar from '@material-ui/core/Avatar';

export default function CustomAvatar(props) {
  return <Avatar {...props} />;
}
