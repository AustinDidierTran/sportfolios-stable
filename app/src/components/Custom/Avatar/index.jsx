import React from 'react';

import { Avatar } from '../../MUI';
import { Icon } from '../../Custom';
import clsx from 'clsx';

import styles from './Avatar.module.css';
import { makeStyles } from '@material-ui/core';

const useStyles = makeStyles({
  avatarNoBackgroundColor: {
    backgroundColor: 'transparent !important',
  },
});

export default function CustomAvatar(props) {
  const { initials, photoUrl, icon, ...otherProps } = props;
  const classes = useStyles();

  let className = clsx(styles.avatar, props.className);
  if (props.size === 'sm') {
    className = clsx(styles.avatar, styles.sm, props.className);
  } else if (props.size === 'md') {
    className = clsx(styles.avatar, styles.md, props.className);
  } else if (props.size === 'lg') {
    className = clsx(styles.lg, styles.avatar, props.className);
  }

  if (photoUrl) {
    return (
      <Avatar
        {...otherProps}
        className={[className, classes.avatarNoBackgroundColor].join(
          ' ',
        )}
        src={photoUrl}
        alt={initials}
      >
        {initials}
      </Avatar>
    );
  }

  if (icon) {
    return (
      <Avatar {...otherProps} className={className} alt={initials}>
        <Icon icon={icon}></Icon>
      </Avatar>
    );
  }

  return (
    <Avatar {...otherProps} className={className}>
      {initials}
    </Avatar>
  );
}
