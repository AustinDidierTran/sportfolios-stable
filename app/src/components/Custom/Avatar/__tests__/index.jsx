import React from 'react';
import Avatar from '../../Avatar';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const props = {
    initials: 'ADT',
    photoUrl: 'photo',
  };

  const tree = renderer.create(<Avatar {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
