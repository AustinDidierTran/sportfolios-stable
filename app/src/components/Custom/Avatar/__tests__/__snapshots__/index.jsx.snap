// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiAvatar-root MuiAvatar-circle avatar"
>
  <img
    alt="ADT"
    className="MuiAvatar-img"
    src="photo"
  />
</div>
`;
